src/Symfony/Component/Ldap/Tests/Adapter/ExtLdap/LdapManagerTest.php

```diff
@@ -113,4 +113,21 @@ private function executeSearchQuery($expectedResults = 1)
 
         return $results;
     }
+
+    /**
+     * @group functional
+     */
+    public function testLdapRename()
+    {
+        $result = $this->executeSearchQuery(1);
+
+        $entry = $result[0];
+
+        $entryManager = $this->adapter->getEntryManager();
+        $entryManager->rename($entry, 'cn=Kevin');
+
+        $result = $this->executeSearchQuery(1);
+        $entry = $result[0];
+        $this->assertEquals($entry->getAttribute('cn')[0], 'Kevin');
```

Comment:
Another thing. You should make sure that the test returns the directory in its original state. If you rename the entry, don't forget to rename it back to its original value after the test.